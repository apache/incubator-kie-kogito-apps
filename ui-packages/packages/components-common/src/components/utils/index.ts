export * from './FormActionsUtils';
export * from './FormValidator';
export * from './ModelConversionTool';
