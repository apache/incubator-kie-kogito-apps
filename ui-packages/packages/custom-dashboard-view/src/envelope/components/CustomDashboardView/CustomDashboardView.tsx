/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useEffect, useRef, useState } from 'react';
import { OUIAProps, componentOuiaProps } from '@kogito-apps/ouia-tools';
import { CustomDashboardViewDriver } from '../../../api/CustomDashboardViewDriver';
import { ServerErrors } from '@kogito-apps/components-common';
import { Bullseye, Card } from '@patternfly/react-core';

export interface CustomDashboardViewProps {
  isEnvelopeConnectedToChannel: boolean;
  driver: CustomDashboardViewDriver;
  customDashboardName: string;
}

const CustomDashboardView: React.FC<CustomDashboardViewProps & OUIAProps> = ({
  isEnvelopeConnectedToChannel,
  driver,
  ouiaId,
  customDashboardName,
  ouiaSafe
}) => {
  const ref = useRef(null);
  const [dashboardContent, setDashboardContent] = useState<string>();
  const [errorMessage, setErrorMessage] = useState<string>();
  const [isError, setError] = useState<boolean>(false);
  const [isReady, setReady] = useState<boolean>(false);
  driver
    .getCustomDashboardContent(customDashboardName)
    .then((value) => setDashboardContent(value))
    .catch((error) => {
      setError(true);
      setErrorMessage(error.message);
    });

  window.addEventListener('message', (e) => {
    if (e.data == 'ready') {
      setReady(true);
    }
  });

  useEffect(() => {
    if (isReady) {
      ref.current.contentWindow.postMessage(dashboardContent, null);
    }
  });

  return (
    <>
      {isError ? (
        <>
          {isEnvelopeConnectedToChannel && (
            <Card className="kogito-custom-dashboard-view-__card-size">
              <Bullseye>
                <ServerErrors error={errorMessage} variant="large" />
              </Bullseye>
            </Card>
          )}
        </>
      ) : (
        <iframe
          ref={ref}
          id="db"
          src="resources/webapp/custom-dashboard-view/dashbuilder/index.html"
          style={{ width: '100%', height: '100%', padding: '10px' }}
          {...componentOuiaProps(ouiaId, 'customDashboard-view', ouiaSafe)}
        />
      )}
    </>
  );
};

export default CustomDashboardView;
