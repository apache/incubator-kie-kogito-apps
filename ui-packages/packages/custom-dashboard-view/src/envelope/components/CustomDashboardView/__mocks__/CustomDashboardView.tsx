import React from 'react';

const MockedCustomDashboardView = (): React.ReactElement => {
  return <></>;
};

export default MockedCustomDashboardView;
