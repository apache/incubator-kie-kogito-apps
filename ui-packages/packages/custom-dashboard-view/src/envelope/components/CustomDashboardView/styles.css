.kogito-custom-dashboard-view-__card-size {
  height: 100%;
}
