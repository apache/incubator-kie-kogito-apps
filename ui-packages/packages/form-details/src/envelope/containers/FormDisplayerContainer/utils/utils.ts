import { Form } from '../../../../api';

import uuidv4 from 'uuid';

export function buildTestContext(form: Form) {
  return {
    user: {
      id: 'test',
      groups: ['group1', 'group2']
    },
    schema: JSON.parse(form.configuration.schema),
    task: {
      id: uuidv4(),
      description: 'task description',
      name: 'TaskName',
      referenceName: 'Task Name',
      priority: '1',
      processInstanceId: uuidv4(),
      processId: 'process',
      rootProcessInstanceId: null,
      rootProcessId: null,
      state: 'Ready',
      completed: null,
      started: Date.now(),
      lastUpdate: Date.now()
    },
    phases: ['abort', 'claim', 'complete', 'skip']
  };
}
