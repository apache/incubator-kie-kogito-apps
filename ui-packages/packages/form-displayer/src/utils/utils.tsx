/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const renderTags = (container, resources): void => {
  for (const key in resources.scripts) {
    const script: HTMLScriptElement = document.createElement('script');

    script.src = resources.scripts[key];
    container.appendChild(script);
  }

  for (const key in resources.styles) {
    const link: HTMLLinkElement = document.createElement('link');
    link.rel = 'stylesheet';
    link.href = resources.styles[key];
    container.appendChild(link);
  }
};

export const renderResources = (containerId, resources): void => {
  const container: HTMLElement = document.getElementById(containerId);
  const scripts: HTMLCollectionOf<HTMLScriptElement> = container.getElementsByTagName(
    'script'
  );
  const styles: HTMLCollectionOf<HTMLLinkElement> = container.getElementsByTagName(
    'link'
  );
  if (scripts.length > 0 || styles.length > 0) {
    let scriptIndex: number = scripts.length;
    let styleIndex: number = styles.length;
    while (scriptIndex--) {
      container.removeChild(scripts[scriptIndex]);
    }
    while (styleIndex--) {
      container.removeChild(styles[styleIndex]);
    }
    renderTags(container, resources);
  } else {
    renderTags(container, resources);
  }
};

export const sourceHandler = source => {
  const reactReg = /import React, {[^}]*}.*(?='react').*/gim;
  const patternflyReg = /import {[^}]*}.*(?='@patternfly\/react-core').*/gim;
  const regexvalueReact = new RegExp(reactReg);
  const reactImport = regexvalueReact.exec(source);
  const reg = /\{([^)]+)\}/;
  const reactElements = reg.exec(reactImport[0])[1];
  const regexvaluePat = new RegExp(patternflyReg);
  const patternflyImport = regexvaluePat.exec(source);
  const patternflyElements = reg.exec(patternflyImport[0])[1];
  const trimmedSource = source
    .split(reactReg)
    .join('')
    .trim()
    .split(patternflyReg)
    .join('')
    .trim();
  const tempSource = trimmedSource;
  const formName = tempSource.split(':')[0].split('const ')[1];

  return { reactElements, patternflyElements, formName, trimmedSource };
};
