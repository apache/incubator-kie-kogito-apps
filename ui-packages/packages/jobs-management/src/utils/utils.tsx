import React from 'react';
import { JobStatus } from '@kogito-apps/management-console-shared/dist/types';
import { UndoIcon } from '@patternfly/react-icons/dist/js/icons/undo-icon';
import { ClockIcon } from '@patternfly/react-icons/dist/js/icons/clock-icon';
import { ErrorCircleOIcon } from '@patternfly/react-icons/dist/js/icons/error-circle-o-icon';
import { BanIcon } from '@patternfly/react-icons/dist/js/icons/ban-icon';
import { CheckCircleIcon } from '@patternfly/react-icons/dist/js/icons/check-circle-icon';

export const JobsIconCreator = (state: JobStatus): JSX.Element => {
  switch (state) {
    case JobStatus.Error:
      return (
        <>
          <ErrorCircleOIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--danger-color--100)"
          />
          Error
        </>
      );
    case JobStatus.Canceled:
      return (
        <>
          <BanIcon className="pf-u-mr-sm" />
          Canceled
        </>
      );
    case JobStatus.Executed:
      return (
        <>
          <CheckCircleIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--success-color--100)"
          />
          Executed
        </>
      );
    case JobStatus.Retry:
      return (
        <>
          <UndoIcon className="pf-u-mr-sm" />
          Retry
        </>
      );
    case JobStatus.Scheduled:
      return (
        <>
          <ClockIcon className="pf-u-mr-sm" />
          Scheduled
        </>
      );
  }
};
