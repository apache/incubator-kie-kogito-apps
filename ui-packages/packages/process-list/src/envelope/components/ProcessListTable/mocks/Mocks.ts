/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
  ProcessInstance,
  ProcessInstanceState
} from '@kogito-apps/management-console-shared';
export const ProcessInstances: ProcessInstance[] = [
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  },
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  }
];
