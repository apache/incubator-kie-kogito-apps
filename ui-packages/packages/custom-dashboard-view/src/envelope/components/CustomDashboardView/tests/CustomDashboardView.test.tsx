/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { mount } from 'enzyme';
import { ServerErrors } from '@kogito-apps/components-common';
import React from 'react';
import CustomDashboardView from '../CustomDashboardView';
import { act } from 'react-dom/test-utils';
import { Bullseye, Card } from '@patternfly/react-core';
import { BrowserRouter } from 'react-router-dom';
import { MockedCustomDashboardViewDriver } from '../../../../embedded/tests/utils/Mocks';
import wait from 'waait';

const props = {
  isEnvelopeConnectedToChannel: true,
  driver: MockedCustomDashboardViewDriver(),
  customDashboardName: 'name'
};
const MockedComponent = (): React.ReactElement => {
  return <></>;
};
jest.mock('@kogito-apps/components-common', () =>
  Object.assign({}, jest.requireActual('@kogito-apps/components-common'), {
    ServerErrors: () => <MockedComponent />
  })
);

describe('Custom Dashboard View tests', () => {
  it('Snapshot tests with data', async () => {
    (props.driver.getCustomDashboardContent as jest.Mock).mockResolvedValue(
      'its a yml file'
    );
    let wrapper;
    await act(async () => {
      wrapper = mount(<CustomDashboardView {...props} />);
      wrapper = wrapper.update().find('CustomDashboardView');
      wait();
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CustomDashboardView)).toMatchSnapshot();
    expect(wrapper.find(CustomDashboardView)).toBeTruthy();
    const iframeWrapper = wrapper.find('iframe');
    expect(iframeWrapper.find('iframe').props()['src']).toEqual(
      'resources/webapp/custom-dashboard-view/dashbuilder/index.html'
    );
    expect(props.driver.getCustomDashboardContent).toHaveBeenCalled();
  });

  it('Snapshot tests with error', async () => {
    (props.driver.getCustomDashboardContent as jest.Mock).mockRejectedValue({
      message: 'network issue'
    });
    let wrapper;
    await act(async () => {
      wrapper = mount(
        <BrowserRouter>
          <CustomDashboardView {...props} />
        </BrowserRouter>
      );
      wrapper = wrapper.update().find('CustomDashboardView');
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(CustomDashboardView)).toMatchSnapshot();
    expect(wrapper.find(CustomDashboardView)).toBeTruthy();
    const cardWrapper = wrapper.update().find(Card);
    expect(cardWrapper.find(Card)).toMatchSnapshot();
    expect(cardWrapper.find(Card).props()['className']).toEqual(
      'kogito-custom-dashboard-view-__card-size'
    );

    const bullseyeWrapper = wrapper.update().find(Card);
    expect(bullseyeWrapper.find(Bullseye)).toMatchSnapshot();

    const serverErrorsWrapper = wrapper.update().find(ServerErrors);
    expect(serverErrorsWrapper.find(ServerErrors)).toMatchSnapshot();
    expect(props.driver.getCustomDashboardContent).toHaveBeenCalled();

    expect(serverErrorsWrapper.find(ServerErrors).props()['error']).toEqual(
      'network issue'
    );
  });
});
