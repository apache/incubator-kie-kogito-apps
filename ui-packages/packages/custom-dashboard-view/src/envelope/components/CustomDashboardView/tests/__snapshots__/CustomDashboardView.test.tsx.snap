// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Dashboard View tests Snapshot tests with data 1`] = `
<CustomDashboardView
  customDashboardName="name"
  driver={
    Object {
      "getCustomDashboardContent": [MockFunction] {
        "calls": Array [
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
    }
  }
  isEnvelopeConnectedToChannel={true}
>
  <iframe
    data-ouia-component-type="customDashboard-view"
    data-ouia-safe={true}
    id="db"
    src="resources/webapp/custom-dashboard-view/dashbuilder/index.html"
    style={
      Object {
        "height": "100%",
        "padding": "10px",
        "width": "100%",
      }
    }
  />
</CustomDashboardView>
`;

exports[`Custom Dashboard View tests Snapshot tests with data 2`] = `
<CustomDashboardView
  customDashboardName="name"
  driver={
    Object {
      "getCustomDashboardContent": [MockFunction] {
        "calls": Array [
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
    }
  }
  isEnvelopeConnectedToChannel={true}
>
  <iframe
    data-ouia-component-type="customDashboard-view"
    data-ouia-safe={true}
    id="db"
    src="resources/webapp/custom-dashboard-view/dashbuilder/index.html"
    style={
      Object {
        "height": "100%",
        "padding": "10px",
        "width": "100%",
      }
    }
  />
</CustomDashboardView>
`;

exports[`Custom Dashboard View tests Snapshot tests with error 1`] = `
<CustomDashboardView
  customDashboardName="name"
  driver={
    Object {
      "getCustomDashboardContent": [MockFunction] {
        "calls": Array [
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
    }
  }
  isEnvelopeConnectedToChannel={true}
>
  <iframe
    data-ouia-component-type="customDashboard-view"
    data-ouia-safe={true}
    id="db"
    src="resources/webapp/custom-dashboard-view/dashbuilder/index.html"
    style={
      Object {
        "height": "100%",
        "padding": "10px",
        "width": "100%",
      }
    }
  />
</CustomDashboardView>
`;

exports[`Custom Dashboard View tests Snapshot tests with error 2`] = `
<CustomDashboardView
  customDashboardName="name"
  driver={
    Object {
      "getCustomDashboardContent": [MockFunction] {
        "calls": Array [
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
          Array [
            "name",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
    }
  }
  isEnvelopeConnectedToChannel={true}
>
  <iframe
    data-ouia-component-type="customDashboard-view"
    data-ouia-safe={true}
    id="db"
    src="resources/webapp/custom-dashboard-view/dashbuilder/index.html"
    style={
      Object {
        "height": "100%",
        "padding": "10px",
        "width": "100%",
      }
    }
  />
</CustomDashboardView>
`;

exports[`Custom Dashboard View tests Snapshot tests with error 3`] = `
<Card
  className="kogito-custom-dashboard-view-__card-size"
>
  <article
    className="pf-c-card kogito-custom-dashboard-view-__card-size"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe={true}
    id=""
  >
    <Bullseye>
      <div
        className="pf-l-bullseye"
      >
        <ServerErrors
          error="network issue"
          variant="large"
        >
          <MockedComponent />
        </ServerErrors>
      </div>
    </Bullseye>
  </article>
</Card>
`;

exports[`Custom Dashboard View tests Snapshot tests with error 4`] = `
<Bullseye>
  <div
    className="pf-l-bullseye"
  >
    <ServerErrors
      error="network issue"
      variant="large"
    >
      <MockedComponent />
    </ServerErrors>
  </div>
</Bullseye>
`;

exports[`Custom Dashboard View tests Snapshot tests with error 5`] = `
<ServerErrors
  error="network issue"
  variant="large"
>
  <MockedComponent />
</ServerErrors>
`;
